swatch-billable-usage/src/main/java/com/redhat/swatch/billable/usage/services/BillableUsageStatusConsumer.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.billable.usage.services;
+
+import com.redhat.swatch.billable.usage.configuration.Channels;
+import com.redhat.swatch.billable.usage.data.BillableUsageRemittanceRepository;
+import com.redhat.swatch.billable.usage.data.RemittanceErrorCode;
+import com.redhat.swatch.billable.usage.data.RemittanceStatus;
+import io.smallrye.reactive.messaging.annotations.Blocking;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.transaction.Transactional;
+import java.util.Objects;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.billable.usage.BillableUsageAggregate;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+
+@Slf4j
+@ApplicationScoped
+@AllArgsConstructor
+public class BillableUsageStatusConsumer {
+
+  private final BillableUsageRemittanceRepository remittanceRepository;
+
+  @Incoming(Channels.BILLABLE_USAGE_STATUS)
+  @Blocking
+  @Transactional
+  public void process(BillableUsageAggregate billableUsageAggregate) {
+    var remittances = remittanceRepository.findByIdIn(billableUsageAggregate.getRemittanceUuids());
+    remittances.forEach(
+        remittance -> {
+          if (Objects.nonNull(billableUsageAggregate.getStatus())) {
+            remittance.setStatus(
+                RemittanceStatus.fromString(billableUsageAggregate.getStatus().value()));
+          } else {
+            log.error(
+                "Billable Usage Status update received with no status, {}:",
+                billableUsageAggregate);
```

Comment:
is this an expected scenario? If so, we should trace the log just once (before doing the query).

Reply:
It's not expected but good point I'll move this before the query.